src/meta/MetaHttpHandler.cpp

```diff
@@ -49,21 +79,53 @@ void MetaHttpHandler::onEOM() noexcept {
                 .status(405, "Method Not Allowed")
                 .sendWithEOM();
             return;
+        case HttpCode::E_ILLEGAL_ARGUMENT:
+            ResponseBuilder(downstream_)
+                .status(400, "Bad Request")
+                .sendWithEOM();
+            return;
         default:
             break;
     }
 
-    // read meta daemon status
-    folly::dynamic vals = getStatus();
-    if (returnJson_) {
-        ResponseBuilder(downstream_)
-            .status(200, "OK")
-            .body(folly::toJson(vals))
-            .sendWithEOM();
+    if (method_ == "status") {
+        folly::dynamic vals = getStatus();
+        if (returnJson_) {
+            ResponseBuilder(downstream_)
+                .status(200, "OK")
+                .body(folly::toJson(vals))
+                .sendWithEOM();
+        } else {
+            ResponseBuilder(downstream_)
+                .status(200, "OK")
+                .body(toStr(vals))
+                .sendWithEOM();
+        }
+    } else if (method_ == "download") {
+        if (auto hadoopHome = std::getenv("HADOOP_HOME")) {
```

Comment:
Why did you need hadoop client installed on meta server?

Reply:
I will check hdfs file number is same with part number in meta service